### PR TITLE
OIDC

### PR DESCRIPTION
**Pull request by omsmith**
Uses OIDC to assume the hub role, rather than relying on the secrets being passed in. Should alleviate any issues of the token expiring while Apply is waiting for approval. Could also wire up disabling rotations of the `+tfapply` role to reduce our load on that end.

### CODE REVIEW

**Comment by j3parker** [comment]
oh baby this is good for a lot of reasons, yeah. If we can reduce the number of tokens we rotate that would be a big deal.

**Review by j3parker — APPROVED** [review]

**Comment by omsmith** [comment]
PR comments are broken, need to figure out the permissions, hmmph

File: .github/workflows/workflow-autoapply.yml [hunk 1]
```diff
@@ -128,6 +122,20 @@ jobs:
...
+      uses: Brightspace/third-party-actions@aws-actions/configure-aws-credentials
+      with:
+        role-to-assume: arn:aws:iam::323258989788:role/hub-roles/github+${{ github.repository_owner }}+${{ github.event.repository.name }}+repo
+        aws-region: us-east-1
+
+    - if: ${{ github.event_name != 'pull_request' }}
+      name: Assume Hub Role
+      uses: Brightspace/third-party-actions@aws-actions/configure-aws-credentials
+      with:
+        role-to-assume: arn:aws:iam::323258989788:role/hub-roles/github+${{ github.repository_owner }}+${{ github.event.repository.name }}+repo
```
Comment (cfbao) [inline .github/workflows/workflow-autoapply.yml:136]: I assume?
```suggestion
        role-to-assume: arn:aws:iam::323258989788:role/hub-roles/github+${{ github.repository_owner }}+${{ github.event.repository.name }}+tfapply
```

Comment (omsmith) [reply .github/workflows/workflow-autoapply.yml]: heh

**Comment by omsmith** [comment]
Comments fixed by 352c4fa

https://github.com/Brightspace/repo-settings/pull/2381
https://github.com/Brightspace/aws-jumpbox/pull/1014

**Review by cfbao — APPROVED** [review]

**Pull request merged**